### PR TITLE
Multi step progress component

### LINKED ISSUE #348: Make progress component
## Explanation

Component looks like this:

Screenshot 2019-09-02 at 12 54 54

Error state:

error

Component has steps.
Steps could be `current` or `done` or `failed`. 

https://sketch.cloud/s/gEP3d/a/M55Lol/play

### PR DESCRIPTION
**Pull request by Xyfi**
## Summary

This PR can be summarized in the following changelog entry:

* Adds a `MultiStepProgress` component.

## Relevant technical choices:

*

## Test instructions

This PR can be tested by following these steps:

* Use the components app.
* Compare to the designs shown in https://github.com/Yoast/javascript/issues/348.
* Remember that the font would look differently in WordPress.

## Impact check

* This PR affects the following parts of the plugin, which may require extra testing:
  * N/A

## UI changes
* [ ] This PR changes the UI in the plugin. I have added the 'UI change' label to this PR.

## Quality assurance

* [x] I have tested this code to the best of my abilities
* [ ] I have added unittests to verify the code works as intended

Fixes #348

### CODE REVIEW

File: packages/components/src/MultiStepProgress.js [hunk 1]
```diff
@@ -0,0 +1,205 @@
...
+
+const MultiStepProgressContainer = styled.div`
+	padding: 8px;
+`;
+
+/**
+ * Ordered list containing the steps.
+ */
+const MultiStepProgressList = styled.ol`
+	padding-left: 0;
```
[the remark this answers is not in the document]
Comment (Xyfi) [reply packages/components/src/MultiStepProgress.js]: Just made it `padding: 0;`

File: packages/components/src/MultiStepProgress.js [hunk 2]
```diff
@@ -0,0 +1,206 @@
...
+		counter-increment: multi-step-progress-counter;
+	}
+`;
+
+/**
+ * List item for a single step.
+ */
+const MultiStepProgressListItem = styled.li`
+	display: flex;
+	flex-direction: row;
```
Comment (afercia) [inline packages/components/src/MultiStepProgress.js]: `flex-direction: row;` is the default value and it's a bit redundant

File: packages/components/src/MultiStepProgress.js [hunk 3]
```diff
@@ -0,0 +1,206 @@
...
+		content: counter( multi-step-progress-counter );
+		font-size: 12px;
+		background: ${ colors.$color_pink_dark };
+		border-radius: 50%;
+		min-width: 16px;
+		height: 16px;
+		padding: 4px;
+		color: ${ colors.$color_white };
+		text-align: center;
+		display: inline-block;
```
Comment (afercia) [inline packages/components/src/MultiStepProgress.js]: this is a flex item so `display: inline-block;` has no effect

File: packages/components/src/SvgIcon.js [hunk 4]
```diff
@@ -1,9 +1,54 @@
...
+ */
+const StyledSvgSpinner = styled.svg`
+	width: ${ props => props.size };
+	height: ${ props => props.size };
+	flex: none;
+
+	animation: loadingSpinnerRotator 1.4s linear infinite;
+
+	& .path {
+		stroke: ${ props => props.color };
```
Comment (afercia) [inline packages/components/src/SvgIcon.js]: Was wondering why passing a `color` prop had no effect. Turns out the prop this component actually receives is `fill`.

File: packages/components/src/SvgIcon.js [hunk 5]
```diff
@@ -1,9 +1,54 @@
...
+		100% {
+			stroke-dashoffset: 187;
+			transform: rotate( 450deg );
+		}
+	}
+`;
+
+StyledSvgSpinner.defaultProps = {
+	color: "#000",
+};
```
Comment (afercia) [inline packages/components/src/SvgIcon.js]: Is this default prop really necessary? Seems to me it already receives `currentColor` as default value. See `<ComponentToUse>`.

**Comment by afercia** [comment]
CR: a few remarks, addressed in the last commit 👍  

Couple notes:

1
In the related issue https://github.com/Yoast/javascript/issues/348 I see the step names were `current` or `done` or `failed`. The actual ones are "pending", "running", "finished", "failed". I guess this was discussed internally?

2
There's no semantic indication of when a step is pending, completed, failed. I'm assuming the actual `text` passed to the steps will clarify the step state otherwise screen reader users won't have a clue what the step status is.

Making the text convey the state info is perfectly fine, just wondering how to make this clear for the implementation.

3
Looking at the sketch https://sketch.cloud/s/gEP3d/a/M55Lol/play I'm not sure automatically opening a modal dialog on page load is the best UI for all users and the best experience for assistive technology users. I guess this should be investigated during implementation.

**Comment by igorschoester** [comment]
To answer for Xyfi:
1. Not really, this is "artistic" freedom. These names are more clear. Plus one was missing.
2. Currently, this does not seem to be needed. If this is needed, the component could be upgraded to HTML support or something along those lines.
3. This is out of scope for this issue.

**Comment by igorschoester** [comment]
Acceptance ✅

**Pull request merged**